File: barricade-annotations/src/main/java/com/mutualmobile/barricade/IBarricadeConfig.java
package com.mutualmobile.barricade;

import com.mutualmobile.barricade.response.BarricadeResponse;
import com.mutualmobile.barricade.response.BarricadeResponseSet;
import java.util.HashMap;

/**
 * Contract for a Barricade configuration
 */
public interface IBarricadeConfig {
  HashMap<String, BarricadeResponseSet> getConfigs();

  BarricadeResponse getResponseForEndpoint(String endpoint);
}

File: barricade-annotations/src/main/java/com/mutualmobile/barricade/IBarricadeConfig.kt
package com.mutualmobile.barricade

import com.mutualmobile.barricade.response.BarricadeResponse
import com.mutualmobile.barricade.response.BarricadeResponseSet

/**
 * Contract for a Barricade configuration
 */
interface IBarricadeConfig {
    fun getConfigs(): HashMap<String, BarricadeResponseSet>
    fun getResponseForEndpoint(endpoint: String): BarricadeResponse?
}
